### ISSUE TITLE
Byteswapped samples (from MAME roms)

### ISSUE THREAD

**Issue by RingoStarr39** [superctr/QuattroPlay#6]
Something about Armadillo Racing (and possibly other games, didn't check) sounds off in quattroplay.
It sounds like the samples are being played at the wrong sample rate or something. It sounds 10 times better in the current version of MAME. While I'm at it, I also updated the ini file for Mr. Driller.
[mrdrillr.txt](https://github.com/superctr/QuattroPlay/files/2413023/mrdrillr.txt)

**Comment by superctr** [comment]
You need to byte swap the sample roms. Unfortunately mame added this in the
rom loading definitions while I proposed just fixing the rom. This is
inconsistent considering that sample roms are 8-bit.

Run the following Unix (msys) commands in the adillor rom directory.
$ cp am1wavea.2l am1wavea.2l_backup
$ cp am1waveb.1l am1waveb.1l_backup
$ dd conv=swab if=am1wavea.2l_backup of=am1wavea.2l bs=1k
$ dd conv=swab if=am1waveb.1l_backup of=am1waveb.1l bs=1k

Den mån 24 sep. 2018 23:52RingoStarr39 <notifications@github.com> skrev:

**Comment by RingoStarr39** [comment]
I noticed the same thing might also be happening with DunkMania. It's samples sound wrong as well.
Byte swapping the roms is outside of my knowledge; is there anyway that functionality could be built into Quattroplay?

**Comment by superctr** [comment]
Dunk Mania is a bad dump. I have the OST, I know what it's supposed to sound like.

Currently it will already byte-swap the program roms for e.g. System 12 games. I don't want to do this for the wave roms though, at least not until the roms are checked with hashes like MAME does.

**Comment by RingoStarr39** [comment]
I just noticed the same thing with Time Crisis. Especially songs that use bell samples. They don't even sound like the same sample.
I have Time Crisis Project Titan for the PlayStation which uses a few tracks from the arcade Time Crisis and they sound much better. You can also tell the difference on the official soundtrack CD.

**Comment by superctr** [comment]
Same thing there, the samples need to be byteswapped. I believe it's only one of the sample roms that are bad in that game.

As I've said, it's really an issue with MAME not having consistent rom loading definitions. Obviously, the original arcade hardware would have all roms in a single format.

**Comment by RingoStarr39** [comment]
Okay, that makes sense. I guess this issue can be closed then.
On a different note, do you plan on doing a new release anytime soon?
It's been almost a year since version 2.0.0 and you've added quite a few new features since then.
I'd compile the changes myself if I knew how, haha. Also, one minor thing about the Mr. Driller ini file.
Track ID 02 is actually an unused version of "Mr.Driller's Theme." The used one is ID 79 and a different version with added percussion is used on a deeper stage which is ID 7A. Also, the rear channels should not be muted in order to match output from the actual PCB.

**Comment by superctr** [comment]
I'll fix the playlist in Mr Driller, in that case. But the rear channels are muted by default as the System 12 hardware is unable to use them (It has only a single LC7881 stereo DAC, while two of them are required for the rear channels).

Also, rough build instructions for windows... (If you already have everything set up skip to step 4)

1. Download MSYS2 from [here](https://www.msys2.org/) and follow the instructions there. It's best to use the default install folder.
2. Open a "mingw64" window (the exe with the blue icon in the installation folder)
3. In this window enter the following:
`pacman -S base-devel mingw-w64-x86_64-toolchain mingw-w64-x86_64-SDL2 git`
4. `git clone https://github.com/superctr/QuattroPlay/`
5. `cd QuattroPlay`
6. `make -j5`
7. to launch the program `./bin/QuattroPlay`. The exe can be found in `<MSYS2 install folder>\home\<blah>\QuattroPlay\bin\`
8. To be able to launch the program from outside the mingw64 shell, `cp /mingw64/bin/SDL2.dll bin`
9. to keep things up to date, navigate to the QP directory, `git pull` and `make -j5` (`make clean` if necessary first)

As a bonus you also have `dd` as part of the MSYS2 tools. You can navigate to your roms directory and use the commands I mentioned earlier to byteswap roms from the mingw64 shell.

**Comment by RingoStarr39** [comment]
Thanks for the instructions, I'll have to try it. Also, I know about System 12 not supporting 4 channel sound but for some reason it sounds more accurate with it turned on. For example, the track "Results / Name Entry" has that synth that pans back and forth. It should be the same volume when panned both hard left and hard right.

**Comment by superctr** [comment]
I assume you mean the volume for channels 06 and 07 in mrdrillr song 08. I've verified that QP's behavior matches what the game's sound driver writes to the C352 registers in MAME. In any case, this should go in another issue.
